### PR TITLE
Add support for using BMP280 with SPI

### LINKED ISSUE #1321: SPI Support for BMP280
**Describe the problem you have/What new integration you would like**

BMP280 sensor supports I2C as well as SPI but ESPHome currently supports only I2C for it. It would be great if SPI support was added for this sensor as well.

**Please describe your use case for this integration and alternatives you've tried:**

Same as above

**Additional context**

Same as above

### PR DESCRIPTION
**Pull request by ademuri**
This adds support for using BMP280 with SPI, similar to #5538.

# What does this implement/fix?

This adds a `bmp280_spi` component, and moves the existing `bmp280` component to `bmp280_i2c`.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes https://github.com/esphome/feature-requests/issues/1321.

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#4064

## Test Environment

- [X] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
sensor:
  - platform: bmp280_spi
    temperature:
      name: "Board Temperature"
      oversampling: 16x
    pressure:
      name: "Board Pressure"
      oversampling: 16x
    update_interval: 5s
    cs_pin: GPIO16

spi:
  clk_pin: GPIO13
  mosi_pin: GPIO15
  miso_pin: GPIO14
  interface: hardware
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/bmp280_base/__init__.py [hunk 1]
```diff
@@ -14,9 +14,9 @@
     CONF_OVERSAMPLING,
 )
 
-DEPENDENCIES = ["i2c"]
+CODEOWNERS = ["@esphome/core"]
```
Comment (jesserockz) [inline esphome/components/bmp280_base/__init__.py]: This is not "owned" by the core team. Either this needs to be owned by you as the person making this PR, or just leave it blank as you are only moving the base code. You can ignore the bot message in this case.

```suggestion
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
Please leave a minimal file in place at `bmp280/__init__.py` and `bmp280/sensor.py` similar to `bmp3xx`

File: esphome/components/bmp280_spi/sensor.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import spi
+from ..bmp280_base import to_code_base, CONFIG_SCHEMA_BASE
+
+AUTO_LOAD = ["bmp280_base"]
+DEPENDENCIES = ["spi"]
```
Comment (jesserockz) [inline esphome/components/bmp280_spi/sensor.py:8]: Not sure why you removed as codeowner on the actual new component you are adding yourself...
```suggestion
CODEOWNERS = ["@ademuri"]
DEPENDENCIES = ["spi"]
```

File: esphome/components/bmp280/sensor.py [hunk 3]
```diff
@@ -1,96 +1,7 @@
...
-    "2X": BMP280IIRFilter.BMP280_IIR_FILTER_2X,
-    "4X": BMP280IIRFilter.BMP280_IIR_FILTER_4X,
-    "8X": BMP280IIRFilter.BMP280_IIR_FILTER_8X,
-    "16X": BMP280IIRFilter.BMP280_IIR_FILTER_16X,
-}
-
-BMP280Component = bmp280_ns.class_(
-    "BMP280Component", cg.PollingComponent, i2c.I2CDevice
-)
+CODEOWNERS = ["@ademuri"]
```
Comment (jesserockz) [inline esphome/components/bmp280/sensor.py]: A codeowner is not needed for removed components
```suggestion
```

File: esphome/components/bmp280/sensor.py [hunk 4]
```diff
@@ -1,96 +1,7 @@
...
-                }
-            ),
-            cv.Optional(CONF_IIR_FILTER, default="OFF"): cv.enum(
-                IIR_FILTER_OPTIONS, upper=True
-            ),
-        }
-    )
-    .extend(cv.polling_component_schema("60s"))
-    .extend(i2c.i2c_device_schema(0x77))
+CONFIG_SCHEMA = CONFIG_SCHEMA = cv.invalid(
```
Comment (jesserockz) [inline esphome/components/bmp280/sensor.py]:
```suggestion
CONFIG_SCHEMA = cv.invalid(
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/bmp280/sensor.py (hunk 3 above)
Comment (ademuri) [reply esphome/components/bmp280/sensor.py]: Done - I mistakenly added it here instead of in `bmp280_spi/sensor.py`.

**Comment by ademuri** [comment]
Apologies for the spam, I accidentally rebased against the wrong dev branch. I did in fact apply all of the requested changes, but Github is showing one of them as outstanding.

File: esphome/components/bmp280_base/__init__.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    CONF_ID,
+    CONF_PRESSURE,
+    CONF_TEMPERATURE,
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TEMPERATURE,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_CELSIUS,
+    UNIT_HECTOPASCAL,
+    CONF_IIR_FILTER,
+    CONF_OVERSAMPLING,
```
Comment (kbx81) [inline esphome/components/bmp280_base/__init__.py]:
```suggestion
    CONF_ID,
    CONF_IIR_FILTER,
    CONF_OVERSAMPLING,
    CONF_PRESSURE,
    CONF_TEMPERATURE,
    DEVICE_CLASS_PRESSURE,
    DEVICE_CLASS_TEMPERATURE,
    STATE_CLASS_MEASUREMENT,
    UNIT_CELSIUS,
    UNIT_HECTOPASCAL,
```

File: tests/components/bmp280_i2c/common.yaml [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
+    i2c_id: i2c_bmp280
     address: 0x77
     temperature:
+      id: bmp280_temperature
       name: Outside Temperature
-      oversampling: 16x
     pressure:
       name: Outside Pressure
-    iir_filter: 16x
+      id: bmp280_pressure
```
Comment (kbx81) [inline tests/components/bmp280_i2c/common.yaml:15]: We should leave `iir_filter: 16x` in as it appears it is still in the config. (Same for SPI config)

**Review by kbx81 — DISMISSED** [review]
Couple other quick remarks

File: tests/components/bmp280_i2c/common.yaml (hunk 6 above)
Comment (ademuri) [reply tests/components/bmp280_i2c/common.yaml:15]: Good point. Done.

**Review by jesserockz — APPROVED** [review]

**Comment by jesserockz** [comment]
:facepalm: Merging dev back in again after approval means I now have to wait for the CI to run again...haha

**Pull request merged**